### DISCUSSION TITLE
Why some Collection methods return a new Collection instance rather than modify self property?

### DISCUSSION THREAD

**Discussion by liquid207** [laravel/framework#38534] [Q&A]
Such as \Illuminate\Support\Collection::flatten()

```
    public function flatten($depth = INF)
    {
        return new static(Arr::flatten($this->items, $depth));
    }
```

This will lead to a problem: 

```
User::all(); // result is `Illuminate\Database\Eloquent\Collection`
User::all()->groupBy('id')->flatten(); // result is `\Illuminate\Support\Collection` 
```
In the chain call, the data type has changed. What is the reason for doing this？

**Comment by MeesterDev** [comment]
FYI the cause of this is not the `flatten()` implementation you've added, but rather the one of the Eloquent Collection, which calls `toBase()`. I am curious what the reasoning for this is.

Seems it was added in commit 91a31e9c377cd43c1658dbc0d86cc737576d7243

**Reply by liquid207** [reply]
This is also the point I am curious about.
